### PR TITLE
add replace enforce with glog switch for debug

### PR DESCRIPTION
**Pull request by Superjomn**
the `PADDLE_ENFORCE`'s exception

image

GLOG error:

image

with gdb, there is 

image

The more clear error stack with line number is provided, easier for debugging.

### CODE REVIEW

**Comment by jacquesqiao** [comment]
Please run with gdb and enforce and see if we can also see the message with line number.

**Comment by Superjomn** [comment]
Run the program in gdb with exception cannot get any call stack calling `bt` command.

File: CMakeLists.txt [hunk 1]
```diff
@@ -131,6 +132,10 @@ if (NOT DEFINED WITH_MKLDNN)
         set(WITH_MKLDNN OFF)
     endif()
 endif()
+
+if (REPLACE_ENFORCE_GLOG)
```
Comment (gongweibao) [inline CMakeLists.txt:136]: put into configure.cmake?

**Review by luotao1 — APPROVED** [review]
LGTM！

File: paddle/fluid/platform/enforce.h [hunk 2]
```diff
@@ -212,6 +238,9 @@ inline void throw_on_error(T e) {
                                               __FILE__, __LINE__);      \
     }                                                                   \
   } while (false)
+#else
+#define PADDLE_ENFORCE(...) ::paddle::platform::throw_on_error(__VA_ARGS__);
```
Comment (gongweibao) [inline paddle/fluid/platform/enforce.h:242]: 这个地方为何去掉了try, catch部分？

File: paddle/fluid/string/printf.h [hunk 3]
```diff
@@ -84,7 +84,7 @@ void Fprintf(std::ostream& out, const char* fmt, const Args&... args) {
 }
 
 template <typename... Args>
-std::string Sprintf(const char* fmt, const Args&... args) {
+std::string Sprintf(const char* fmt = "", const Args&... args) {
```
Comment (gongweibao) [inline paddle/fluid/string/printf.h:87]: 默认参数一般是最后一个吧？

**Review by gongweibao — CHANGES_REQUESTED** [review]

**Review by gongweibao — APPROVED** [review]

**Pull request merged**